src/baseline.rs

```diff
@@ -340,3 +360,104 @@ fn need_retry(res: Result<(), crates_index::Error>) -> anyhow::Result<bool> {
         Err(err) => Err(err.into()),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crates_index::{Crate, Version};
+
+    use super::choose_baseline_version;
+
+    fn new_mock_version(version_name: &str, yanked: bool) -> Version {
+        // `crates_index::Version` cannot be created explicitly, as all its fields
+        // are private, so we use the fact that it can be deserialized.
+        serde_json::from_value(serde_json::json!({
+            "name": "test-crate",
+            "vers": version_name,
+            "deps": [],
+            "features": {},
+            "yanked": yanked,
+            "cksum": "00".repeat(32),
+        }))
+        .expect("Parsing JSON as crates_index::Version should not result in any errors.")
```

Comment:
Something I learned from @obi1kenobi -- this message describes an invariant, but it makes more sense in an error message to describe what failed. So maybe consider writing it as `Failed to create crates_index::Version from a hand-written JSON`.